### ISSUE TITLE
Create proper CLI with subcommands

### ISSUE THREAD

**Issue by senderista** [uwescience/myria-ec2-ansible#3]
We should have a consistent CLI for controlling/configuring Myria EC2 clusters analogous to the `spark-ec2` scripts (https://spark.apache.org/docs/0.9.1/ec2-scripts.html, https://github.com/amplab/spark-ec2). It might look something like this:

```
myria-cluster create [cluster-name] [options]
myria-cluster destroy [cluster-name] [options]
myria-cluster login [cluster-name] [options]
myria-cluster describe [cluster-name] [options]
myria-cluster modify [cluster-name] [options]
myria-cluster suspend [cluster-name] [options]
myria-cluster resume [cluster-name] [options]
myria-cluster list [options]
myria-cluster [subcommand] help
myria-cluster help
```

We probably should implement this in Python rather than a shell script to keep the options parsing/handling manageable. This might mean switching the EC2 API calls to `boto` rather than the AWS CLI, and running the Ansible playbook from the Ansible Python API (http://docs.ansible.com/ansible/developing_api.html, https://github.com/ansible/ansible/blob/release1.8.4/lib/ansible/playbook/__init__.py#L42-L84). We could also implement our own internal API/object model to keep the script to a simple "driver" model, but no reason to overengineer if we're the only consumers of the API.

**Comment by BrandonHaynes** [comment]
`myria-cluster verify-aws-permissions`?

**Comment by senderista** [comment]
Sure, or maybe just `verify` (could also verify local installation prerequisites etc.).

**Comment by senderista** [comment]
This should be installed as a proper python package (using `pip` or `conda`). That would let the packaging infrastructure handle dependency tracking for us. Note that `setuptools` will generate an executable script for us in the user's path if we specify an entry point.

**Comment by BrandonHaynes** [comment]
Random idea: `myria-cluster repl` would be useful too, where we could automatically launch an interactive Python session with a default connection already established along with other convenience functionality.

**Comment by senderista** [comment]
@BrandonHaynes did you have something like this in mind? https://github.com/ipython/ipython/wiki/Cookbook:-Connecting-to-a-remote-kernel-via-ssh

**Comment by BrandonHaynes** [comment]
Not sure we need anything so sophisticated just loading up a python shell and initializing with something like `MyriaRelation.DefaultConnection = MyriaConnection('http://...')` could be nice.  Something like the way Spark does it with its context instance.

**Comment by senderista** [comment]
Got it, no remote interpreter, just a python shell preloaded with the `myria-python` module, initialized with our cluster's connection info. No reason to require IPython to be installed, right? Sounds like a great UX feature.

**Comment by senderista** [comment]
fixed in b3176792e1b3f92b69a8f3858bda906aefc57b33
@BrandonHaynes if you're still interested in the `myria-cluster repl` idea, please open a separate issue and assign to me.
